### ISSUE TITLE
Checkout block: if Stripe is not fully configured, other payment methods are not shown

### ISSUE THREAD

**Issue by haszari** [woocommerce/woocommerce-gutenberg-products-block#3125]
## Describe the bug
If the store has Stripe enabled, but no API key set, there's a JS error and no payment methods are displayed (breaks others e.g. BACS).

<img width="825" alt="Screen Shot 2020-09-09 at 9 35 37 AM" src="https://user-images.githubusercontent.com/4167300/92530110-d403ab00-f27f-11ea-9979-9bcf498025ef.png">

<img width="887" alt="Screen Shot 2020-09-09 at 9 33 18 AM" src="https://user-images.githubusercontent.com/4167300/92530084-ca7a4300-f27f-11ea-8873-8a1e106f647c.png">

## To reproduce
1. Install & activate WooCommerce Stripe. 
2. Enable Stripe CC payment method - don't add an api key (or delete the option).
3. Enable BACS or other payment methods (e.g. cheque).
3. Add checkout block to checkout page.
4. On front end, add something to cart and proceed to checkout.

### Expected behavior
- Expected: valid configured payment methods should be shown, and if there are none, the "no payment methods" error should be displayed.
- Actual: JS error breaks the flow, no payment methods available, no error message.

Also it looks like a bug in one payment method can break other methods! We should ensure that there's enough try/catch protection so each payment method code can't break other methods or checkout generally.

### WordPress (please complete the following information):
- WC Version: 4.4.1
- WC Blocks Version: 3.2, 3.3.0-dev

<details>
   <summary>WooCommerce status report</summary>

```
### WordPress Environment ###

WordPress address (URL): https://ephemeral-haszari-20200831.atomicsites.blog
Site address (URL): https://ephemeral-haszari-20200831.atomicsites.blog
WC Version: 4.4.1
REST API Version: ✔ 1.0.15
WC Blocks Version: ✔ 3.2
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.4.0
Log Directory Writable: ✔
WP Version: ❌ 5.5 - There is a newer version of WordPress available (5.5.1)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.3.22
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.72.0
OpenSSL/1.1.0l

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.21-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.4.1
WC Database Prefix: wp_
Total Database Size: 2.50MB
Database Data Size: 1.09MB
Database Index Size: 1.41MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wp_posts: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 23
page: 7
post: 1
product: 18
product_variation: 7
revision: 6
shop_order: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (5) ###

Akismet Anti-Spam: by Automattic – 4.1.6
Jetpack by WordPress.com: by Automattic – 8.8.2
WooCommerce Blocks: by Automattic – 3.2
WooCommerce Stripe Gateway: by WooCommerce – 4.5.2 – Installed version not tested with active version of WooCommerce 4.4.1
WooCommerce: by Automattic – 4.4.1

### Inactive Plugins (0) ###


### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: ❌ Page does not contain the shortcode.
Checkout: ❌ Page does not contain the shortcode.
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty
Version: 1.5
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 9
Oldest: 2020-08-31 23:55:16 +0000
Newest: 2020-09-08 21:31:13 +0000

Pending: 1
Oldest: 2020-09-09 21:30:07 +0000
Newest: 2020-09-09 21:30:07 +0000

```

</details>

**Comment by haszari** [comment]
Spotted when testing #3121 against 3.2. Will confirm if this is also present in 3.3.0.

**Comment by haszari** [comment]
Reproduced in current dev build (3.3.0-dev).

<img width="1553" alt="Screen Shot 2020-09-09 at 9 47 23 AM" src="https://user-images.githubusercontent.com/4167300/92530974-88ea9780-f281-11ea-9cfd-fedd9c8ee74d.png">

**Comment by haszari** [comment]
Comparison behaviour with shortcode checkout - Stripe is not available, other payment methods are shown:

<img width="958" alt="Screen Shot 2020-09-09 at 9 48 26 AM" src="https://user-images.githubusercontent.com/4167300/92531012-9ef85800-f281-11ea-8648-20731d5ed0c3.png">

Block should behave in a similar way to shortcode.

**Comment by Aljullu** [comment]
Good find with this issue @haszari! I created a PR that should fix it in #3134, but in order to keep this issue updated as well: if I'm not wrong, the bug is only reproducible if you are logged as an admin user, so it's not affecting most shoppers.

Pointing this out in case we decide to do a 3.3 point release, I don't think a fix for this issue needs to be included.
